pkg/store/gcworker/gc_worker.go

```diff
@@ -880,9 +910,31 @@ func (w *GCWorker) deleteRanges(ctx context.Context, safePoint uint64, concurren
 	return nil
 }
 
+const (
+	// ConcurrencyDivisor reduces the input concurrency to avoid overwhelming the system
+	ConcurrencyDivisor = 4
+	// RequestsPerThread is the number of requests handled by a single thread
+	RequestsPerThread = 100000
+
+	// Assuming an average request takes 50ms:
+	// With ideal parallelism and sufficient concurrency,
+	// the maximum duration for a round of deleteRanges is 100,000 * 50ms = 5,000s.
+	// These values are conservatively chosen to minimize GC impact on foreground requests.
+)
+
+func (w *GCWorker) calcDeleteRangeConcurrency(concurrency gcConcurrency, n int) int {
+	maxConcurrency := concurrency.v / ConcurrencyDivisor
```

Comment:
Could you add some explanations about why we should have two concurrencies of `gcConcurrency` and `deleteRangeConcurrency` and why they have this relationship?

Reply:
For example, for placement GC gc, all HTTP calls are sent to one PD endpoint which is a hotspot and if the `concurrency.v` is very big, we still cannot ensure it returns a proper value even if using a `ConcurrencyDivisor` to divide it.

Reply:
I added some. 
> if the concurrency.v is very big, we still cannot ensure it returns a proper value

The algorithm is proposed by PM @joey-yez. Do you have any suggestion?